ui/src/actionBar.tsx

```diff
@@ -7,14 +7,20 @@ import Typography from "@mui/material/Typography";
 import { useUnderlay } from "hooks";
 import * as React from "react";
 import { Link as RouterLink } from "react-router-dom";
+import { useBackUrl } from "router";
+import { Underlay } from "underlaysSlice";
 
 type ActionBarProps = {
   title: string;
-  backUrl?: string;
 };
 
 export default function ActionBar(props: ActionBarProps) {
-  const underlay = useUnderlay();
+  let underlay: Underlay | undefined;
+  try {
+    underlay = useUnderlay();
+  } catch {}
```

Comment:
Did you mean to leave this in? TODO or logging to handle this? If it's truly needed I would add a comment - blanket exception supression like this can obfuscate programming errors

Reply:
You're right, this is confusing and asking for trouble down the road. This is the one place an undefined underlay is potentially expected. Instead I've included the code to fetch the underlay directly since that seems better than a separate function that doesn't throw plus a comment explaining why it shouldn't generally be used.